src/modules/peek/Peek.UI/FileQuery.cs

```diff
@@ -0,0 +1,170 @@
+                {
+                    Debug.WriteLine("Detected unexpected existing initializeFilesTask running. Cancelling it..");
+                    cancellationTokenSource.Cancel();
+                }
+
+                cancellationTokenSource = new CancellationTokenSource();
+                initializeFilesTask = new Task(() => InitializeFiles(items, firstSelectedItem, cancellationTokenSource.Token));
+
+                // Execute file initialization/querying on background thread
+                initializeFilesTask.Start();
+            }
+            catch (Exception e)
+            {
+                Debug.WriteLine("Exception trying to run initializeFilesTask:\n" + e.ToString());
+            }
+        }
+
+        // Finds index of firstSelectedItem either amongst folder items, initializing our internal File list
+        //  since storing Shell32.FolderItems as a field isn't reliable.
+        // Can take a few seconds for folders with 1000s of items; ensure it runs on a background thread.
+        //
+        // TODO optimization:
+        //  Handle case where selected items count > 1 separately. Although it'll still be slow for 1000s of items selected,
+        //  we can leverage faster APIs like Windows.Storage when 1 item is selected, and navigation is scoped to
+        //  the entire folder. We can then avoid iterating through all items here, and maintain a dynamic window of
+        //  loaded items around the current item index.
+        private void InitializeFiles(
+            Shell32.FolderItems items,
+            Shell32.FolderItem firstSelectedItem,
+            CancellationToken cancellationToken)
+        {
+            var tempFiles = new List<File>(items.Count);
+            var tempCurIndex = UninitializedItemIndex;
+
+            for (int i = 0; i < items.Count; i++)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var item = items.Item(i);
+                if (item == null)
+                {
+                    continue;
+                }
+
+                if (item.Name == firstSelectedItem.Name)
+                {
+                    tempCurIndex = i;
+                }
+
+                tempFiles.Add(new File(item.Path));
+            }
+
+            if (tempCurIndex == UninitializedItemIndex)
+            {
+                Debug.WriteLine("File query initialization: selectedItem index not found. Navigation remains disabled.");
+                return;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            lock (mutateQueryDataLock)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                files = tempFiles;
+                currentItemIndex = tempCurIndex;
+            }
+        }
+
+        private readonly object mutateQueryDataLock = new ();
+
+        [ObservableProperty]
+        private File? currentFile;
+
+        private List<File> files = new ();
```

Comment:
nit: use properties for all of these fields instead. It is easier to debug and put them at the top of the file.

Reply:
Capital letter for first letter as well. It adds clarity when looking for properties vs temp var in the code.